client/settings/express-checkout-settings/payment-request-button-preview.js

```diff
@@ -103,88 +152,84 @@ const PaymentRequestButtonPreview = () => {
-									style: {
-										paymentRequestButton: {
-											type: buttonType,
-											theme: theme,
-											height: `${
-												buttonSizeToPxMap[ size ] ||
-												buttonSizeToPxMap.medium
-											}px`,
-										},
-									},
-								} }
-							/>
-						) }
-				</div>
-			) }
-			{ ! isWooPayEnabled && ! isPaymentRequestEnabled && (
-				<InlineNotice icon status="info" isDismissible={ false }>
-					{ __(
-						'To preview the express checkout buttons, ' +
-							'activate at least one express checkout.',
-						'woocommerce-payments'
-					) }
-				</InlineNotice>
-			) }
-			{ isPaymentRequestSettingsPage() &&
-				isPaymentRequestEnabled &&
-				! isLoading &&
-				! paymentRequest && (
-					<InlineNotice icon status="info" isDismissible={ false }>
-						{ __(
-							'To preview the Apple Pay and Google Pay buttons, ' +
-								'ensure your device is configured to accept Apple Pay or Google Pay, ' +
-								'and view this page using the Safari or Chrome browsers.',
-							'woocommerce-payments'
-						) }
-					</InlineNotice>
+	// No need to check `isStripeEceEnabled` since that's not what controls whether the express checkout
+	// buttons are displayed or not, that's always controlled by `isPaymentRequestEnabled`.
+	if ( ! isWooPayEnabled && ! isPaymentRequestEnabled ) {
+		return (
+			<InlineNotice icon status="info" isDismissible={ false }>
+				{ __(
+					'To preview the express checkout buttons, ' +
+						'activate at least one express checkout.',
+					'woocommerce-payments'
 				) }
-			<BrowserHelpText />
-		</>
-	);
+			</InlineNotice>
+		);
+	}
+
+	const woopayPreview = isWooPayEnabled ? (
+		<WooPayButtonPreview
+			size={ size }
+			buttonType={ buttonType }
+			theme={ theme }
+			radius={ radius }
+		/>
+	) : null;
+
+	const expressCheckoutButtonPreview =
+		isPaymentRequestEnabled && isStripeEceEnabled ? (
+			<ExpressCheckoutPreviewComponent
+				stripe={ stripe }
+				buttonType={ buttonType }
+				theme={ theme }
+				height={ buttonSizeToPxMap[ size ] || buttonSizeToPxMap.medium }
+				radius={ radius }
+			/>
+		) : null;
+
+	const prbButtonPreview =
+		isPaymentRequestEnabled && paymentRequest && ! isLoading ? (
+			<PaymentRequestButtonElement
+				key={ `${ buttonType }-${ theme }-${ size }` }
+				onClick={ ( e ) => {
+					e.preventDefault();
+				} }
+				options={ {
+					paymentRequest: paymentRequest,
+					style: {
+						paymentRequestButton: {
+							type: buttonType,
+							theme: theme,
+							height: `${
+								buttonSizeToPxMap[ size ] ||
+								buttonSizeToPxMap.medium
+							}px`,
+						},
+					},
+				} }
+			/>
+		) : null;
+
+	if ( woopayPreview || expressCheckoutButtonPreview || prbButtonPreview ) {
+		return (
+			<ButtonPreviewWrapper theme={ theme }>
+				{ woopayPreview }
+				{ /* We never want to show both ECE and PRB previews at the same time. */ }
+				{ ( window.location.protocol === 'https:' &&
```

Comment:
This check is preventing the [inline notice](https://github.com/Automattic/woocommerce-payments/pull/9028/files#diff-8368d420c86a67fc8fc3bceafa24455eeaf836e4dca6e0c1de3da227cae67604R97) from being displayed in case the settings is accessed through an insecure connection. I believe we can remove it in favor of the notice.

One thing to note is that the inline notice may also be displayed when user is customizing the WooPay button. I guess this is expected (and desired)? cc @pierorocca

Reply:
Ah, sorry, I forgot to update the testing instructions here. You need to remove this check for the `Make sure an info notice is displayed when no button can be rendered` test to work.

The check is there to give the merchant better information on why the button might not be rendering. The test is to check the more generic fallback in the ECE preview component that could, in theory, be triggered by other failures. Trying to open the page on HTTP is just the only way I know to force the buttons to fail rendering 😅 

So yeah, maybe you can try re-running that test after removing the `window.location.protocol === 'https:'` check?

Reply:
I've also updated the testing instructions 👍

Reply:
I understand what you're trying to achieve here, and the test case was correct previously. What I'm calling out is that the notice will never be displayed if the user is on an insecure connection because the current logic doesn't allow that. What do you think about moving that check inside the component to prevent the buttons to render and enable the notice in such cases, maybe to the [onReady callback](https://github.com/Automattic/woocommerce-payments/pull/9028/files#diff-8368d420c86a67fc8fc3bceafa24455eeaf836e4dca6e0c1de3da227cae67604R71-R77)?

Reply:
It actually does show up in Firefox:

<img width="706" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/4e79d5b5-69fd-4679-80f6-a70b68927579">

Should've just made that the test case, don't know why I didn't think of that before.

But yeah, it is absolutely necessary when the merchant tries to look at the preview in a browser that isn't configured correctly or doesn't support Apple Pay or Google Pay.

Reply:
Moving the `https` check into the component would cause the notice not to appear when ECE is disabled, because the notice catches the `HTTPS` condition for both ECE and PRBs.

Reply:
> One thing to note is that the inline notice may also be displayed when user is customizing the WooPay button. I guess this is expected (and desired)? cc @pierorocca

I _think_ this is desired since the WooPay button appears in the GPay/APay customization menu? It makes sense to me that the same should apply in the WooPay customization menu so you can see what your checkout will actually look like.

Reply:
> I think this is desired since the WooPay button appears in the GPay/APay customization menu? It makes sense to me that the same should apply in the WooPay customization menu so you can see what your checkout will actually look like.

That logic makes sense to me @reykjalin. When we get to refreshing the payment settings and consolidating where we can, this won't be as confusing of an experience.

Reply:
>maybe you can try re-running that test after removing the window.location.protocol === 'https:' check?

I don't think removing that check to test this is the right call because we can simulate it being `true` or `false` without removing it.

>It actually does show up in Firefox:

I don't see the notice in Firefox when testing through a non-HTTPS connection.

<img width="1189" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/10233985/15b83d3f-27b9-4c7e-a3d9-93c79cc2f3fe">


>Moving the https check into the component would cause the notice not to appear when ECE is disabled, because the notice catches the HTTPS condition for both ECE and PRBs.

That's true. Then, we can tweak the logic to pass its value over to the component instead of conditionally render it and prevent the notice from rendering. The `https` check has been introduced in this PR, so I'm confident we can restructure it to please all our needs 🙂 Unless we have a strong reason not to do so?

I really would like this test case to be successful because I'm sure there will be folks accessing this page via HTTP and the fact that they don't see the ECE/PRB button neither a notice may cause confusion.

Reply:
> I don't see the notice in Firefox when testing through a non-HTTPS connection.

Ah, this was actually a bug! Or sort of intended? Basically to only show the WooPay button when that's the only button available. I fixed this in 245091b785d41f0c39b73346bf2b74b8abe5e4b2 though.

<details><summary>A chart of what the various settings now look like:</summary>

| Browser | HTTPS? | WooPay? | APay/GPay? | Screenshot |
|:---:|:---:|:---:|:---:|:---:|
| Firefox | ⛔️ | ⛔️ | ⛔️ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/ff830df1-834b-4105-8bf5-70a28fcd9e83"> |
| Firefox | ⛔️ | ✅ | ⛔️ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/1e9fceba-63fb-49d5-b48d-47f48ac2ec50"> |
| Firefox | ⛔️ | ⛔️ | ✅ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/b72e02f5-c344-4e34-80cd-f1414fcb5604"> |
| Firefox | ⛔️ | ✅ | ✅ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/b175c6b2-c98c-48bd-b1d7-aa454f5c7355"> |
| Firefox | ✅ | ⛔️ | ⛔️ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/a8487dd9-ee7e-4d27-b2ab-9b83ef1f69be"> |
| Firefox | ✅ | ✅ | ⛔️ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/16cf4e89-8ac1-4cac-8900-fee3e84ca49c"> |
| Firefox | ✅ | ⛔️ | ✅ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/ae4db014-a326-4076-9a07-e593f224a7a6"> |
| Firefox | ✅ | ✅ | ✅ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/ea49462c-d2b7-4db8-b1ed-765b6a6684f1"> |
| Chrome | ⛔️ | ⛔️ | ⛔️ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/1468e983-520b-49a4-9db2-7067e101cf10"> |
| Chrome | ⛔️ | ✅ | ⛔️ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/3fa92e35-a471-4fa0-a7fc-1b40d0674e74"> |
| Chrome | ⛔️ | ⛔️ | ✅ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/f62b616a-445f-4d6f-a309-c39404b910b9"> |
| Chrome | ⛔️ | ✅ | ✅ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/38b533e7-8409-43b8-b751-60f46fd69c72"> |
| Chrome | ✅ | ⛔️ | ⛔️ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/651c66f5-b759-490b-b55e-8378d428b23d"> |
| Chrome | ✅ | ✅ | ⛔️ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/ddc6914e-41f9-4c55-83b8-994348232466"> |
| Chrome | ✅ | ⛔️ | ✅ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/0ef22a89-2e11-4cb7-b75e-ad1d9d851ac6"> |
| Chrome | ✅ | ✅ | ✅ | <img width="716" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/13835680/8df73fa5-f80b-45ac-bd1f-2048df95044d"> |

</details>

> Then, we can tweak the logic to pass its value over to the component instead of conditionally render it and prevent the notice from rendering. The https check has been introduced in this PR, so I'm confident we can restructure it to please all our needs 🙂 Unless we have a strong reason not to do so?

We _could_ but I don't really like it because we'd need to either duplicate the notice, or include it from another component. Or pass it in as some sort of `fallbackNotice` prop on both the PRB and ECE preview components.

The nice thing about the if-statements in the `<PaymentRequestButtonPreview />` parent component is that it's easy to find all the components it renders instead of drilling down into each component. I think it feels better, architecturally speaking, to keep the requirements info notice in the `<PaymentRequestButtonPreview />` component instead of passing more state down to the `<ExpressCheckoutPreviewComponent />` and `<PaymentRequestButtonElement />` components.

> I really would like this test case to be successful because I'm sure there will be folks accessing this page via HTTP and the fact that they don't see the ECE/PRB button neither a notice may cause confusion.

This should be addressed by the latest commit 👍

Reply:
That looks great 🎉 Thank you for addressing this, Kris!